### ISSUE TITLE
patrol_finders doesn't work with latest Flutter beta

### ISSUE THREAD

**Issue by bartekpacia** [leancodepl/patrol#1834]
That's because of changes made in https://github.com/flutter/flutter/pull/127137. Finder API was changed.

When that `beta` becomes the latest `stable`, we'll have to bump Patrol's minimum Flutter version to that. I'm not aware of any other solutions to this problem.

cc @mateuszwojtczak @jBorkowska

**Comment by bartekpacia** [comment]
I fixed the compile-time errors, but new run-time errors appeared:

```

10-30 13:23:27.650  9058  9096 I flutter : Patrol (native): markPatrolAppServiceReady() succeeded
10-30 13:23:27.677  9058  9096 I flutter : ══╡ EXCEPTION CAUGHT BY FLUTTER FRAMEWORK ╞═════════════════════════════════════════════════════════
10-30 13:23:27.677  9058  9096 I flutter : The following assertion was thrown:
10-30 13:23:27.677  9058  9096 I flutter : The render object for Stack cannot find ancestor render object to attach to.
10-30 13:23:27.677  9058  9096 I flutter : The ownership chain for the RenderObject in question was:
10-30 13:23:27.677  9058  9096 I flutter :   Stack ← [root]
10-30 13:23:27.677  9058  9096 I flutter : Try wrapping your widget in a View widget or any other widget that is backed by a
10-30 13:23:27.677  9058  9096 I flutter : RenderTreeRootElement to serve as the root of the render tree.
10-30 13:23:27.677  9058  9096 I flutter : ════════════════════════════════════════════════════════════════════════════════════════════════════
10-30 13:23:27.688  9058  9096 I flutter : ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
10-30 13:23:27.688  9058  9096 I flutter : The following assertion was thrown building View:
10-30 13:23:27.688  9058  9096 I flutter : The RenderObject for _RawView-[_DeprecatedRawViewKey TestFlutterView#1e2f5] cannot maintain an
10-30 13:23:27.688  9058  9096 I flutter : independent render tree at its current location.
10-30 13:23:27.688  9058  9096 I flutter : The ownership chain for the RenderObject in question was:
10-30 13:23:27.688  9058  9096 I flutter :   _RawView-[_DeprecatedRawViewKey TestFlutterView#1e2f5] ← View ← RepaintBoundary ← Stack ← [root]
10-30 13:23:27.688  9058  9096 I flutter : This RenderObject is the root of an independent render tree and it cannot attach itself to an
10-30 13:23:27.688  9058  9096 I flutter : ancestor in an existing tree. The ancestor RenderObject, however, expects that a child will be
10-30 13:23:27.688  9058  9096 I flutter : attached.
10-30 13:23:27.688  9058  9096 I flutter : Try moving the subtree that contains the _RawView-[_DeprecatedRawViewKey TestFlutterView#1e2f5]
10-30 13:23:27.688  9058  9096 I flutter : widget into the view property of a ViewAnchor widget or to the root of the widget tree, where it is
10-30 13:23:27.688  9058  9096 I flutter : not expected to attach its RenderObject to a parent.
10-30 13:23:27.688  9058  9096 I flutter : 
10-30 13:23:27.688  9058  9096 I flutter : The relevant error-causing widget was:
10-30 13:23:27.688  9058  9096 I flutter :   RepaintBoundary



```

**Comment by bartekpacia** [comment]
Fixed on #1829.

**Comment by richchurcher** [comment]
I don't suppose there's a particular combination of versions I can use which allows Flutter beta channel while also using Patrol, while we wait? I tried using your branch:
```
  patrol:
    git:
      path: packages/patrol
      ref: task/patrol-extension
      url: https://github.com/leancodepl/patrol
  patrol_finders:
    git:
      path: packages/patrol_finders
      ref: task/patrol-extension
      url: https://github.com/leancodepl/patrol
```
but the dependency on `patrol_finders ^1.0.0` stopped me (with Flutter `3.16.0-0.5.pre`).

> Because bucket depends on patrol from git which depends on patrol_finders ^1.0.0, patrol_finders from hosted is required.                                                                                            
> So, because bucket depends on patrol_finders from git, version solving failed.

**Comment by muhleder** [comment]
@richchurcher Have you tried using a dependency override for patrol_finders?

**Comment by richchurcher** [comment]
That's exactly what I did yesterday, ty! I hadn't discovered overrides yet 🙂 🙏

**Comment by bartekpacia** [comment]
Closed in #1829.
